faker/providers/date_time/__init__.py

```diff
@@ -20,6 +21,8 @@
 
 def datetime_to_timestamp(dt):
     if getattr(dt, 'tzinfo', None) is not None:
+        if getattr(dt, 'tzinfo', 'UTC'):
```

Comment:
Hi, I've checked this issue one more time. And I see error in test described in issue #513 in some cases.
I found that I see this bug for my timezone (+3) for years from 1930 till 1989. Seems it's related with "decret time"(timezone time +1 hour) which was implemented in USSR from 1930.
So instead using `mktime` I propose to convert datetime to UTC, and then use `timegm` to convert to timestamp.
If it's acceptable, I can make pull request.
Thx.